homeassistant/components/livebox/__init__.py

```diff
@@ -0,0 +1,149 @@
+        manufacturer=config["Manufacturer"],
+        name=config["ProductClass"],
+        model=config["ModelName"],
+        sw_version=config["SoftwareVersion"],
+    )
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "sensor")
+    )
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "binary_sensor")
+    )
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "device_tracker")
+    )
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+
+    await hass.config_entries.async_forward_entry_unload(config_entry, "sensor")
+    await hass.config_entries.async_forward_entry_unload(config_entry, "binary_sensor")
+    await hass.config_entries.async_forward_entry_unload(config_entry, "device_tracker")
+    return True
+
+
+class LiveboxData:
+    """Collect datas information from livebox."""
+
+    def __init__(self, config_entry):
+        """Init datas from router."""
+
+        self._entry = config_entry
+
+    async def async_devices(self, device=None):
+        """Get devices datas."""
+
+        parameters = {"parameters": {"expression": {"wifi": "wifi"}}}
+        if device is not None:
+            parameters = {
+                "parameters": {
+                    "expression": {
+                        "wifi": 'wifi and .PhysAddress=="' + device["PhysAddress"] + '"'
+                    }
+                }
+            }
+        box = await self.async_conn()
+        devices = (await box.system.get_devices(parameters))["status"]["wifi"]
+
+        return devices
+
+    async def async_infos(self):
+        """Get infos."""
+
+        box = await self.async_conn()
+        infos = (await box.system.get_deviceinfo())["status"]
+
+        return infos
+
+    async def async_status(self):
+        """Get status."""
+
+        box = await self.async_conn()
+        status = (await box.system.get_WANStatus())["data"]
+
+        return status
+
+    async def async_dsl_status(self):
+        """Get dsl status."""
+
+        box = await self.async_conn()
+        parameters = {"parameters": {"mibs": "dsl", "flag": "", "traverse": "down"}}
+        dsl_status = (await box.connection.get_data_MIBS(parameters))["status"]["dsl"][
+            "dsl0"
+        ]
+
+        return dsl_status
+
+    async def async_conn(self):
+        """Connect at the livebox router."""
+
+        box = Sysbus()
+        try:
+            await box.open(
+                host=self._entry.data["host"],
+                port=self._entry.data["port"],
+                username=self._entry.data["username"],
+                password=self._entry.data["password"],
+            )
+            return box
+
+        except AuthorizationError:
+            _LOGGER.error("User or password incorrect")
+            return False
```

Comment:
This is an inconsistent return type. We should return None in this case.
```suggestion
            return None
```